plugins/woocommerce/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -13,6 +13,71 @@ class BlockifiedProductDetails extends AbstractBlock {
 	 */
 	protected $block_name = 'blockified-product-details';
 
+	/**
+	 * Create accordion item block markup
+	 *
+	 * @param string $title Accordion Title.
+	 * @param string $content Accordion Content.
+	 * @return array
+	 */
+	private function create_accordion_item( $title, $content ) {
+		$markup_html = sprintf(
+			'<!-- wp:woocommerce/accordion-item {"openByDefault": false} --><div class="wp-block-woocommerce-accordion-item"><!-- wp:woocommerce/accordion-header -->
+		<h3 class="wp-block-woocommerce-accordion-header accordion-item__heading"><button class="accordion-item__toggle"><span>%1$s</span><span class="accordion-item__toggle-icon has-icon-plus" style="width:1.2em;height:1.2em"><svg width="1.2em" height="1.2em" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true"><path d="M11 12.5V17.5H12.5V12.5H17.5V11H12.5V6H11V11H6V12.5H11Z" fill="currentColor"></path></svg></span></button></h3>
+		<!-- /wp:woocommerce/accordion-header -->
+		<!-- wp:woocommerce/accordion-panel -->
+		<div class="wp-block-woocommerce-accordion-panel"><div class="accordion-content__wrapper"><!-- wp:html -->%2$s<!-- /wp:html --></div></div><!-- /wp:woocommerce/accordion-panel --></div><!-- /wp:woocommerce/accordion-item --></div>',
+			$title,
+			$content
+		);
+		return parse_blocks( $markup_html )[0];
+	}
+
+
+	/**
+	 * Add new accordion item to the parsed block
+	 *
+	 * @param array $parsed_block Parsed block.
+	 * @param array $parsed_tabs_added_via_hook Parsed tabs added via hook.
+	 */
+	private function add_new_accordion_item( &$parsed_block, $parsed_tabs_added_via_hook ) {
+		foreach ( $parsed_block['innerBlocks'] as &$inner_block ) {
+			if ( 'woocommerce/accordion-group' === $inner_block['blockName'] ) {
+					$inner_block['innerBlocks'] = array_merge( $inner_block['innerBlocks'], $parsed_tabs_added_via_hook );
+				foreach ( $parsed_tabs_added_via_hook as $block ) {
+					// It is necessary update the innerContent given that a new inner block is added.
+					array_splice(
+						$inner_block['innerContent'],
+						-1,
+						0,
+						array( "\n\n", null )
+					);
+				}
+			}
+			self::add_new_accordion_item( $inner_block, $parsed_tabs_added_via_hook );
```

Comment:
I don't quite understand why we are calling this recursively? should this be within an `else`?

Reply:
I added a comment to clarify its necessity: df899b42e094cfc7ebc12b1768b7d49d27c3af4d.  

Also, two things that I want to clarify:
- The backend logic should support multiple accordion groups. The client-side will enforce the "single group" limitation. This could be useful in the future, if we will decide to remove the limitation.
- To achieve this, all inner blocks of Product Details must be explored. Recursion should stop only when `$inner_blocks` is empty, ensuring the entire tree is traversed.